services/blockstorage/sync/block_sync_test.go

```diff
@@ -0,0 +1,45 @@
+package sync
+
+import (
+	"github.com/orbs-network/go-mock"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func TestBlockSyncShutdown(t *testing.T) {
+	h := newBlockSyncHarness()
+
+	h.storage.When("LastCommittedBlockHeight").Return(primitives.BlockHeight(10)).Times(1)
+	h.gossip.When("BroadcastBlockAvailabilityRequest", mock.Any).Return(nil, nil).Times(1)
+	sync := NewBlockSync(h.ctx, h.config, h.gossip, h.storage, h.logger)
+	h.cancel()
+	time.Sleep(time.Millisecond)
+	// TODO: refactor this once more logic is added, this is not really checking the the goroutine stopped
+	require.True(t, sync.terminated, "expecting the stop flag up")
+}
+
+func TestBlockSyncStaysInIdleOnBlockCommitExternalMessage(t *testing.T) {
```

Comment:
it covers HandleBlockCommitted()
there is another unit test inside idleState that covers the code there and then another one at a higher level covering the flow from gossip - felt that instead of relying on the gossip test that will find problem anywhere in the pipeline, to have all of these tests so its easier to pinpoint the problem

Reply:
my point was - since this test duplicates other tests at lower levels of the system, you should document why this test is needed

Reply:
my point was - since this test duplicates other tests at lower levels of the system, you should document why this test is needed

Reply:
missed that point :) i'll document it